content/spire/try/oidc-federation-aws.md

```diff
@@ -0,0 +1,415 @@
+
+In this tutorial you will learn how to:
+
+* Deploy the OIDC Discovery Provider Service
+* Create the required DNS A record to point to the Discovery Provider document
+* Create a sample AWS identity provider, policy, role, and S3 bucket
+* Test access to the S3 bucket
+
+## Prerequisites
+
+Note the following required accounts, prerequisites, and limitations before starting this tutorial:
+
+* You'll need access to the Kubernetes environment that you configured when going through [Kubernetes Quickstart](/spire/try/getting-started-k8s/) 
+* You'll need access to the AWS console to configure an identity provider, policy, role, and S3 bucket
+* This tutorial cannot run on Minikube as the Kubernetes environment must be network accessible by AWS
+* You'll need the ability to configure a DNS A record for the SPIRE OIDC Discovery Provider document location (see next section)
+
+### Required DNS A Record for the OIDC Discovery Provider IP Address
+
+The SPIRE OIDC Discovery Provider provides a URL to the location of the discovery document specified by the OIDC protocol. After starting the spire-oidc service you'll need to configure a DNS A record to point to the external IP address of that service. In the YAML files set up for this tutorial, replace MY\_DISCOVERY\_DOMAIN with the FQDN of the planned DNS hostname. The FQDN value for MY\_DISCOVERY\_DOMAIN does not need to correspond to an existing domain on your network. It is specific to the Kubernetes environment.
+
+To get the external IP address required for the A record, you will need to proceed through the steps in this tutorial until you start the spire-oidc service and verify the external IP address in the section [Verify That spire-oidc has an External Service Address](#verify-that-spireoidc-has-an-external-service-address). The DNS A record should take the following form:
+
+```console
+MY_DISCOVERY_DOMAIN          A        EXTERNAL-IP for spire-oidc service
+```
+
+For example:
+```console
+oidc-discovery.example.org   A        93.184.216.34
+```
+
+{{< info >}}
+As with any change to DNS, it will take minutes or hours for the new A record to propagate to DNS servers. This tutorial will not work until the A record is propagated. Once the A record is propagated, you should be able to view the discovery document at https://MY_DISCOVERY_DOMAIN/.well-known/openid-configuration .
+{{< /info >}}
+
+Although this tutorial should contain all the info you need, the SPIRE OIDC Discovery Provider configuration file format is described on [GitHub](https://github.com/spiffe/spire/tree/master/support/oidc-discovery-provider).
+
+This tutorial uses Let's Encrypt to configure a certificate for the OIDC Discovery Provider domain. But other than editing the oidc-dp-configmap.yaml file as described in the previous section, no additional setup steps are required for Let's Encrypt.
```

Comment:
Is a Lets Encrypt account a pre-requisite? If so it should be called out at the beginning in the pre-requisites section.

Reply:
No Let's Encrypt account is needed per Eugene. Added "or account" to last sentence here. One could argue that this paragraph could be removed altogether since no user action is required.